alfalfa_worker/worker_openstudio/add_site.py

```diff
@@ -0,0 +1,296 @@
+        self.site_ref = None
+
+    def main(self):
+        """
+        Main entry point after init.  Adds site based on file ext.  Worfklow is generally as follows:
+        1. Download file from s3 bucket
+        2. Ingest model file and add tags
+        3. Send data to mongo and s3 bucket
+        4. Remove files generated during this process
+        :return:
+        """
+        if self.file_ext == '.osm':
+            self.add_osm()
+        elif self.file_ext == '.zip':
+            self.add_osw()
+        elif self.file_ext == '.fmu':
+            self.add_fmu()
+        else:
+            self.add_site_logger.logger.error("Unsupported file extension: {}".format(self.file_ext))
+            os.exit(1)
+
+    def extract_workflow_tar(self):
+        """
+        Extract workflow tarball into this directory
+        :return:
+        """
+        tar = tarfile.open("workflow.tar.gz")
+        tar.extractall(self.bucket_parsed_site_id_dir)
+        tar.close()
+
+    def get_site_ref(self, haystack_json):
+        """
+        Find the site given the haystack JSON file.  Remove 'r:' from string.
+        :param haystack_json: json serialized Haystack document
+        :return: site_ref: id of site
+        """
+        site_ref = ''
+        with open(haystack_json) as json_file:
+            data = json.load(json_file)
+            for entity in data:
+                if 'site' in entity:
+                    if entity['site'] == 'm:':
+                        site_ref = entity['id'].replace('r:', '')
+                        break
+        return site_ref
+
+    def insert_fmu_tags(self, points_json_path):
+        with open(points_json_path, 'r') as f:
+            data = f.read()
+        points_json = json.loads(data)
+
+        self.ac.add_site_to_mongo(points_json, self.upload_id)
+
+    def insert_os_tags(self, points_json_path, mapping_json_path):
+        """
+        Make unique ids and replace site_id.  Upload to mongo and filestore.
+        :return:
+        """
+        # load mapping and points json files generated by previous workflow
+        with open(points_json_path, 'r') as f:
+            data = f.read()
+        points_json = json.loads(data)
+
+        with open(mapping_json_path, 'r') as f:
+            data = f.read()
+        mapping_json = json.loads(data)
+
+        # fixup tags
+        # This is important to avoid duplicates in the case when a client submits the same model
+        # more than one time
+        points_json, mapping_json = make_ids_unique(points_json, mapping_json)
+        points_json = replace_site_id(self.upload_id, points_json)
+
+        # save "fixed up" json
+        with open(points_json_path, 'w') as fp:
+            json.dump(points_json, fp)
+
+        with open(mapping_json_path, 'w') as fp:
+            json.dump(mapping_json, fp)
+
+        # add points to database
+        self.ac.add_site_to_mongo(points_json, self.upload_id)
+
+    def add_osm(self):
```

Comment:
Is this OSM function purely internal? or am I going to get rid of it soon?

Reply:
OSM is external too and yes, we want to get rid of it. It will be a nice code cleanup!